code/game/objects/structures/tables_racks.dm

```diff
@@ -294,6 +299,59 @@
 	..()
 	return SECONDARY_ATTACK_CONTINUE_CHAIN
 
+/obj/structure/table/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if(!istype(user) || !user.can_interact_with(src) || isobserver(user))
+		return
+	if(!can_flip)
+		return
+	user.balloon_alert_to_viewers("flipping table...")
+	if(!do_after(user, max_integrity * 0.25))
+		return
+
+	var/obj/structure/flippedtable/flipped_table = new flipped_table_type(loc)
+	flipped_table.name = "flipped [name]"
+	flipped_table.desc = "[desc]<br> It's been flipped on its side!"
+	flipped_table.icon_state = base_icon_state
+	var/new_dir = get_dir(user, flipped_table)
+	flipped_table.dir = new_dir
+	if(new_dir == NORTH)
+		flipped_table.layer = BELOW_MOB_LAYER
+	flipped_table.max_integrity = max_integrity
+	flipped_table.update_integrity(get_integrity())
+	flipped_table.table_type = type
+	if(istype(src, /obj/structure/table/greyscale)) //Greyscale tables need greyscale flags!
+		flipped_table.material_flags = MATERIAL_EFFECTS | MATERIAL_COLOR
+	//Finally, add the custom materials, so the flags still apply to it
+	flipped_table.set_custom_materials(custom_materials)
+
+	var/sound_volume = 100
+	var/balloon_message = "table flipped"
+	var/user_pacifist = HAS_TRAIT(user, TRAIT_PACIFISM)
+
+	if (user_pacifist)
+		balloon_message = "table gently flipped"
+		sound_volume = 40
+
+	user.balloon_alert_to_viewers(balloon_message)
```

Comment:
you don't need a balloon alert here because the visual feedback is the table flipping over